missioncontrol/api/views.py

```diff
@@ -46,31 +49,37 @@ def channel_platform_summary(request):
     platform_filter = [platform.lower() for platform in request.GET.getlist('platform')]
     channel_filter = [channel.lower() for channel in request.GET.getlist('channel')]
 
+    platforms = Platform.objects.all()
+    if platform_filter:
+        platforms = platforms.filter(name__in=platform_filter)
+
+    channels = Channel.objects.all()
+    if channel_filter:
+        channels = channels.filter(name__in=channel_filter)
+
     summaries = []
-    for channel_name in filter(lambda c: not channel_filter or c in channel_filter,
-                               CHANNELS.keys()):
-        for (platform_name, platform) in PLATFORMS.items():
-            if not platform_filter or platform_name in platform_filter:
-                measures = []
-                measure_name_map = {
-                    get_measure_summary_cache_key(platform_name, channel_name,
-                                                  measure_name): measure_name
-                    for measure_name in platform['measures']
-                }
-                measure_summaries = cache.get_many(measure_name_map.keys())
-                for (measure_summary_cache_key, measure_summary) in measure_summaries.items():
-                    measures.append({
-                        **{'name': measure_name_map[measure_summary_cache_key]},
-                        **measure_summary,
-                        **{'lastUpdated': _dt_to_utc(measure_summary['lastUpdated'])
-                           if measure_summary.get('lastUpdated') else None}
-                    })
-                summaries.append({
-                    'channel': channel_name,
-                    'platform': platform_name,
-                    'measures': measures
+    for channel in channels:
+        for platform in platforms:
+            measures = []
+            measure_name_map = {
+                get_measure_summary_cache_key(platform.name, channel.name,
+                                              measure_name): measure_name
+                for measure_name in Measure.objects.filter(
+                        platform=platform).values_list('name', flat=True)
+            }
+            measure_summaries = cache.get_many(measure_name_map.keys())
+            for (measure_summary_cache_key, measure_summary) in measure_summaries.items():
+                measures.append({
+                    **{'name': measure_name_map[measure_summary_cache_key]},
+                    **measure_summary,
+                    **{'lastUpdated': _dt_to_utc(measure_summary['lastUpdated'])
+                       if measure_summary.get('lastUpdated') else None}
```

Comment:
I'm intrigued by the syntax here. Isn't this doing the same thing as creating a dictionary like `{'name': measure_name_map[measure_summary_cache_key], ...}` ?

Reply:
Hmm, yes, I'm not actually sure what I was thinking now :) I'll remove the unnecessary complication here.